### ISSUE TITLE
aria-required-children: be less permissive of divs as child nodes

### ISSUE THREAD

**Issue by WilcoFiers** [dequelabs/axe-core#3462]
ARIA required context role, failed example 3

In the [ARIA required context role](https://w3.org/WAI/standards-guidelines/act/rules/ff89c9/proposed) ACT rules, [failed example 3](https://www.w3.org/WAI/content-assets/wcag-act-rules/testcases/ff89c9/52508dc0ac389108301d7cbd7f931be45a45741f.html) currently isn't being failed by axe-core. Specifically, I think aria-required-children should have caught this:

```html
<div role="list">
  <div aria-live="polite">
    <div role="listitem">List item 1</div>
    <div role="listitem">List item 2</div>
  </div>
</div>
```

Because of its global attribute, that `div[aria-live]` should not be ignored in the accessibility tree. How browsers do this varies. Firefox considers this a `section` inside a `list`.

**Comment by straker** [comment]
Should fail due to #3468, just need to add to the `getOwnedRoles` function to check for tabindex > -1 or is focusable (should we make this it's own function called `includedInAccessibilityTree`? We do this type of check often enough it might make sense). Adding `null` to the list or return roles should be sufficient in this case to trigger the error message.

**Comment by padmavemulapati** [comment]
Validated with the latest develop branch code base,
`aria-required-children` failing now for the child elements which are unlisted

<img width="1328" alt="image" src="https://user-images.githubusercontent.com/47409579/187247064-c6687b4b-ca2a-4ab7-ba57-256d518e2536.png">
 
and with 4.3.5 we are reproducing the issue
